providers/user_apps_autostart.rb

```diff
@@ -16,37 +17,65 @@
 #    if new_resource.support_os.include?(os)
     if new_resource.support_os.include?($gecos_os)
       users = new_resource.users 
-      desktop_path = "/usr/share/applications/"
 
+      case node['platform']
+        when 'debian', 'ubuntu', 'redhat', 'centos', 'fedora'
+          applications_path = "/usr/share/applications/"
+          subdirs = %w(.config/ autostart/)
+          file_ext  = '.desktop'
+        when 'windows'
+          # TODO
+        when 'mac_os_x'
+          # TODO
+      end
+          
       users.each_key do |user_key|
         nameuser = user_key 
         username = nameuser.gsub('###','.')
         user = users[user_key]
+        gid  = Etc.getpwnam(username).gid
+
+        autostart_path = ::File.expand_path("~#{username}")
 
-        homedir = `eval echo ~#{username}`.gsub("\n","")
-        autostart_path = "#{homedir}/.config/autostart/"
-        unless Kernel::test('d', autostart_path)
-          FileUtils.mkdir_p(autostart_path)
-          gid = Etc.getpwnam(username).gid
-          FileUtils.chown_R(username, gid, homedir+"/.config")
+        subdirs.each do |subdir|
```

Comment:
You can create dirs recursively with the standar chef resource 
https://stackoverflow.com/questions/35952170/chef-create-directories-recursively-with-owner

Anyway, if you use resources but do not use signals, probably this recipe try to create a file in a not-created-yet directory

Reply:
https://docs.chef.io/resource_directory.html

recursive
Ruby Types: TrueClass, FalseClass

Create or delete parent directories recursively. For the owner, group, and mode properties, **the value of this attribute applies only to the leaf directory**. Default value: false.

Permissions not applied recursively.